service/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package com.epam.esm.service.impl;
+
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dto.TagDto;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.exceptions.TagNotFoundException;
+import com.epam.esm.exceptions.TagWithSuchNameAlreadyExists;
+import com.epam.esm.service.TagService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of {@link com.epam.esm.service.TagService} interface.
+ *
+ * @since 1.0
+ */
+@Service
+@Transactional(propagation = Propagation.SUPPORTS, isolation = Isolation.READ_COMMITTED, readOnly = true)
```

Comment:
Please explain why READ COMMITTED?
And also if I'm not wrong it's default isolation level

Reply:
And why propagation is SUPPORTS?

Reply:
Commit c725a00